### PR TITLE
Small updates to README

### PR DESCRIPTION
**Pull request by akikoskinen**
Update build badge. Remove obsolete stuff. Change API documentation link.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,27 +181,10 @@ The tests require a Postgres database to which to connect to. Here's one way to
 * Run tests within the Django container: `docker-compose exec django pytest`.
 
 
-## Issue tracking
```
Comment (mikkokeskinen) [inline README.md:184]: These shouldn't be removed. See [City of Helsinki documentation practises](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/OS/pages/7738589197/Dokumentaatiok+yt+nn+t+documentation+practices).

**Review by mikkokeskinen — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (akikoskinen) [reply README.md:184]: Ah yes, link to issue tracking. But the documentation practices also says that under "Links to public information", so I guess the non-public link can be removed?

Comment (mikkokeskinen) [reply README.md:184]: Yes, I would assume so.

Comment (akikoskinen) [reply README.md:184]: According to the guidelines publicly accessible environments should also be listed, so I think I'll resurrect (and correct) those too.

**Review by mikkokeskinen — APPROVED** [review]

**Pull request merged**